recipes/boost/all/conanfile.py

```diff
@@ -141,25 +208,73 @@ def configure(self):
                 if not self.options.get_safe('without_%s' % lib):
                     raise ConanInvalidConfiguration("Boost '%s' library requires multi threading" % lib)
 
-        if self.settings.compiler == "Visual Studio" and "MT" in self.settings.compiler.runtime and self.options.shared:
+        if self.settings.compiler == "Visual Studio" and "MT" in str(self.settings.compiler.runtime) and self.options.shared:
             raise ConanInvalidConfiguration("Boost can not be built as shared library with MT runtime.")
 
+        # Check, when a boost module is enabled, whether the boost modules it depends on are enabled as well.
+        for mod_name, mod_deps in self._dependencies["dependencies"].items():
+            if not self.options.get_safe("without_{}".format(mod_name), True):
+                for mod_dep in mod_deps:
+                    if self.options.get_safe("without_{}".format(mod_dep), False):
+                        raise ConanInvalidConfiguration("{} requires {}: {} is disabled".format(mod_name, mod_deps, mod_dep))
+
+        if not self.options.without_python:
+            if not self.options.python_version:
```

Comment:
The default is a string `"None"` does this have the expected behavior?

Reply:
It *worked* for me when building with `conan create . boost/1.74.0@ -o boost:without_python=False`.
So I guess: "yes".